app.js

```diff
@@ -100,6 +100,70 @@ function createApp(config) {
     crawlerSecret
   )
 
+  // ===================================================
```

Comment:
Could we refactor this into a separate file responsible for logging memory usage to improve modularity and clarity, and then invoke it here?  This can be done as a separate enhancement PR.

Reply:
Refactored the heap logging code into a new module called `heapLogger` under the `./lib` directory